src/program/lwaftr/run/run.lua

```diff
@@ -141,18 +141,25 @@ function parse_args(args)
    end
 end
 
+-- Requires a V4V6 splitter iff:
+--   Always when running in on-a-stick mode, except if v4_vlan_tag != v6_vlan_tag.
+local function requires_splitter (opts, conf)
+   if not opts["on-a-stick"] then return false end
+   if not conf.vlan_tagging then return true end
+   return conf.v4_vlan_tag == conf.v6_vlan_tag
+end
+
 function run(args)
    local opts, conf_file, v4, v6 = parse_args(args)
    local conf = require('apps.lwaftr.conf').load_lwaftr_config(conf_file)
+   local use_splitter = requires_splitter(opts, conf)
 
    local c = config.new()
    if opts.virtio_net then
       setup.load_virt(c, conf, 'inetNic', v4, 'b4sideNic', v6)
    elseif opts["on-a-stick"] then
-      setup.load_on_a_stick(c, conf, 'v4v6', {
-         mirror = opts.mirror,
-         pciaddr = v4,
-      })
+      setup.load_on_a_stick(c, conf, 'inetNic', 'b4sideNic',
+                            use_splitter and 'v4v6', v4, opts.mirror)
```

Comment:
Seven positional arguments are a bit to many to follow: maybe keeping the `args` table would be clearer.